agent/connect/ca/provider.go

```diff
@@ -22,6 +22,15 @@ type Provider interface {
 	// ActiveIntermediate()
 	ActiveRoot() (string, error)
 
+	// GenerateIntermediateCSR generates a CSR for an intermediate CA
+	// certificate, to be signed by the root of another datacenter. If isRoot is
+	// true, calling this is an error.
+	GenerateIntermediateCSR() (string, error)
+
+	// SetIntermediate sets the provider to use the given intermediate certificate
+	// as well as the root it was signed by.
```

Comment:
Maybe another comment about isRoot and/or that this completes initialisation for a isRoot = false provider?